pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -189,6 +218,15 @@ func (op *objectPatch) apply(ctx context.Context, c client.Client) error {
 		return nil
 	}
 
+	// Patch() will modify op.obj, which is an object that was
+	// generated from our PmemCSIDeployment object and shares some
+	// data structure with it. We don't want those to be modified,
+	// so here we have to do a deep copy first.
+	copy, err := cloneObject(op.obj)
+	if err != nil {
+		return fmt.Errorf("internal error: %v", err)
+	}
+	op.obj = copy
```

Comment:
I'm not happy with the whole `objectPatch` struct here. It's very hard to understand what it does and now `apply` even has side effects on the content of the `objectPatch` struct itself.

It's only used in one place (`redeploy`) and I am tempted to roll it back into that.

@avalluri what do you think?